modules/ROOT/pages/client-id-based-policies.adoc

```diff
@@ -14,48 +14,73 @@ endif::[]
 
-= Analytics and  access tracking
-If you need to track client application access to your API, apply a policy that directly or indirectly makes sure that client ID is provided in the HTTP request (ie Client ID enforcement type policy).
+// How does the policy work?
+//Where can you get the inforamtion for the policy--Mariano to send me the information
 
-= Indirect client ID enforcement
-There are other provided policies that enforce client application credentials. For example:
 
-* xref:rate-limiting-and-throttling-sla-based-policies.adoc[Rate Limiting - SLA-Based Policies]
+Default policies that internally enforce client application credentials include:
 
+* xref:rate-limiting-and-throttling-sla-based-policies.adoc[Rate Limiting - SLA-Based Policies]
++
 In Mule 3.8.5 or later, the following OAuth 2.0 Token Enforcement policies include a client ID enforcement validation:
-
 * xref:external-oauth-2.0-token-validation-policy.adoc[Mule OAuth 2.0 Access Token Enforcement]
 * xref:policy-ping-federate.adoc[PingFederate OAuth 2.0 Token Enforcement]
 * xref:policy-openid-connect.adoc[OpenID Connect OAuth 2.0 Token Enforcement]
 
-= Offline mode
+== Policy Prerequisites and Restrictions
 
-Credentials data from client application credentials registered to protected APIs, are provided by Anypoint Platform. Once a policy requiring such information is applied (eg Client ID enforcement policy), the Runtime caches persistently the information locally updating it periodically.
+For the policy to work successfully, you must:
 
-Besides its performance benefits, this action enables the policy to work offline in the event of a sudden loss of connectivity to the platform. The enforcement is applied regardless of the state of the management plane, with existing data. Once connectivity is re-established, the cache will be periodically updated again.
+* Generate the client ID and client secret credentials from Anypoint Platform.
++
+Using client credentials generated outside of Anypoint Platform fails the HTTP request.
+* Provide the client ID and client secret pair of the registered client application.
+* If using REST console, create a trait in the RAML root and then reference it in every operation of your API affected by policy.
++
+To obtain the RAML code to add to the policy, go to the corresponding policy in your API Manager.
+//It looks like we might need this for other policies too? If so, we should move it out of here and put it in a central place.
++
+image::raml-snippet.png[]
 
-= REST Console
-To prevent user requests from being rejected when using the REST Console, create a trait in the RAML root and then reference it in every operation of your API affected by policy. You may find the RAML snippet link containing the RAML code you need to add to the RAML in the corresponding policy from the list of applied policies in the Policies tab of your API specification in API Manager.
 
-image::raml-snippet.png[]
+== Policy Implementation Best Practices
```

Comment:
Agree, I wouldn't have a section just for this now.

Reply:
Done! Thanks!